apiserver/facades/client/application/application.go

```diff
@@ -230,6 +218,7 @@ type DeployApplicationFunc = func(
 	context.Context,
 	ApplicationDeployer,
 	Model,
```

Comment:
Is there no way to remove this at the moment?

Reply:
Not at the moment. This is passed into `assertCharmAssumptions` and then into `stateenvirons.SupportedFeatures`. We need to move `stateenvirons.SupportedFeatures` over to using model info/config/agent services instead of state models. I think that is getting to be a bit too much for this PR though.

Reply:
ok fair